Mathlib/Geometry/Manifold/Instances/UnitsOfNormedAlgebra.lean

```diff
@@ -70,4 +51,44 @@ variable {𝕜 : Type*} [NontriviallyNormedField 𝕜] [NormedAlgebra 𝕜 R]
 instance : SmoothManifoldWithCorners 𝓘(𝕜, R) Rˣ :=
   openEmbedding_val.singleton_smoothManifoldWithCorners 𝓘(𝕜, R)
 
+/-- For a complete normed ring `R`, the embedding of the units `Rˣ` into `R` is a smooth map between
+manifolds. -/
+lemma contMDiff_val {m : WithTop ℕ} : ContMDiff 𝓘(𝕜, R) 𝓘(𝕜, R) m (val : Rˣ → R) :=
+  contMDiff_openEmbedding 𝓘(𝕜, R) Units.openEmbedding_val
+
+/-- For any map `f` from a manifold `M` to the units `Rˣ` of a complete normed ring `R`, the
+smoothness of `val ∘ f`, where `val : Rˣ → R` is the embedding, implies the smoothness of `f`.
+
+This can be used to show that ring multiplication `Rˣ × Rˣ → Rˣ` and inverse `Rˣ → Rˣ` are
+smooth. -/
+lemma ContMDiff.of_comp_units_val
+    {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+    {H : Type*} [TopologicalSpace H] {I : ModelWithCorners 𝕜 E H}
+    {M : Type*} [TopologicalSpace M] [ChartedSpace H M]
+    {n : WithTop ℕ} {f : M → Rˣ} (hf : ContMDiff I 𝓘(𝕜, R) n ((val : Rˣ → R) ∘ f)) :
+    ContMDiff I 𝓘(𝕜, R) n f :=
+  ContMDiff.of_comp_openEmbedding Units.openEmbedding_val hf
+
+/-- The units of a complete normed ring form a Lie group. -/
+instance : LieGroup 𝓘(𝕜, R) Rˣ where
+  smooth_mul := by
+    apply ContMDiff.of_comp_units_val
+    have : (val : Rˣ → R) ∘ (fun x : Rˣ × Rˣ => x.1 * x.2) =
+      (fun x : R × R => x.1 * x.2) ∘ (fun x : Rˣ × Rˣ => (x.1, x.2)) := by ext; simp
+    rw [this]
+    have : ContMDiff (𝓘(𝕜, R).prod 𝓘(𝕜, R)) 𝓘(𝕜, R × R) ∞
+      (fun x : Rˣ × Rˣ => ((x.1 : R), (x.2 : R))) :=
+      (contMDiff_val.comp contMDiff_fst).prod_mk_space (contMDiff_val.comp contMDiff_snd)
+    refine' ContMDiff.comp _ this
```

Comment:
I [asked](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/refine.20vs.20refine'/near/403550755) and see a mild preference for `refine`, hence just pushed a commit replacing it. Hope you don't mind; feel free to revert otherwise!